### ISSUE TITLE
Bitrate go brrr? NVENC bug.

### ISSUE THREAD

**Issue by Originalimoc** [LizardByte/Sunshine#1877]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

When you not move the mouse, bitrate is constant at what set(40Mbps here, task manager network showing too), when you just move mouse cursor around, bitrate go whatever it want(shoot to 70-80Mbps and stay there if you keep moving the cursor)? Reproduced with this almost full screen https://www.youtube.com/watch?v=ubFq-wV3Eic

### Expected Behavior

Bitrate always to constrained to max set in Moonlight

### Additional Context

nvidia dedicated desktop one monitor
nightly
ml 5.0.1
HW Encoding working, in HwInfo X Task manager

### Host Operating System

Windows

### Operating System Version

11

### Architecture

64 bit

### Sunshine commit or version

nightly as of writing

### Package

Windows - installer

### GPU Type

Nvidia

### GPU Model

RTX Turing 31.0.15.4601 driver

### GPU Driver/Mesa Version

546.01

### Config

```Shell
nvenc_preset = 2
fps = [30,60,90,120,165,144]
resolutions = [
    1280x720,
    1920x1080,
    1920x1200,
    3840x2160,
    2560x1440,
    1280x800,
    2560x1600
]
fec_percentage = 0
min_threads = 4
qp = 30
encoder = nvenc
address_family = both
channels = 1
av1_mode = 1
```

### Relevant log output

```shell
No
```

**Comment by Originalimoc** [comment]
Client even tried SW decoder, same. AMD iGPU, Intel iGPU, both both

**Comment by Originalimoc** [comment]
Even if you do not play anything and set resolution to 2k144 then move your cursor then bitrate goes from 4Mbps to 30Mbps....

**Comment by Originalimoc** [comment]
@ReenigneArcher @cgutman This is clearly a bug somewhere, weird part is that it only manifests when moving cursor. When set Moonlight to 123Mbps it will shoot to 180-200Mbps.

**Comment by Originalimoc** [comment]
@ReenigneArcher @cgutman Imagine someone with mobile plan not realize it because maybe 5G bandwidth won't bottleneck him but then hit with double the bandwidth usage bill. Also closing issue without any explanation is just silent rude.

**Comment by ReenigneArcher** [comment]
No logs, no bug report

**Comment by Originalimoc** [comment]
Because I checked log, not relevant. Just a bunch of these:
```
[2023:12:01:22:22:11]: Info: Active GPU has HAGS enabled
[2023:12:01:22:22:11]: Info: Using realtime GPU priority
[2023:12:01:22:22:11]: Info: Desktop resolution [2560x1440]
[2023:12:01:22:22:11]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:12:01:22:22:11]: Info: Display refresh rate [143.912Hz]
[2023:12:01:22:22:11]: Info: Requested frame rate [144fps]
[2023:12:01:22:22:11]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.679688,0.313477]
Green Primary      : [0.283203,0.648438]
Blue Primary       : [0.147461,0.0712891]
White Point        : [0.3125,0.329102]
Min Luminance      : 0.1447 nits
Max Luminance      : 426.86 nits
Max Full Luminance : 426.86 nits
[2023:12:01:22:22:11]: Info: Adjusted capture rate to 143.912fps to better match display
[2023:12:01:22:22:11]: Info: Gamepad 0 will be Xbox 360 controller (default)
[2023:12:01:22:22:11]: Info: SDR color coding [Rec. 601]
[2023:12:01:22:22:11]: Info: Color depth: 10-bit
[2023:12:01:22:22:11]: Info: Color range: [MPEG]
[2023:12:01:22:22:11]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2023:12:01:22:22:11]: Info: NvEnc: created encoder P2 10-bit two-pass rfi
```

**Comment by Originalimoc** [comment]
Reopen or I open a new one? Does it have debug/trace log mode?

**Comment by ReenigneArcher** [comment]
Thanks for providing some logs... more might be helpful. Do not open another issue.

**Comment by Originalimoc** [comment]
Controller playing game bitrate also seems within limit, bitrate go wild just if you use mouse....? What a weird bug.

**Comment by ReenigneArcher** [comment]
Can you do this?

1. Do a screen record of you moving your mouse, while not streaming.
2. Play the video on your host, full screen.
3. Start streaming to moonlight.

Is the issue the same? That will indicate if it's related to the mouse movement, or to the image changing rapidly.

**Comment by Originalimoc** [comment]
yes, play what you actually see(the video) is within limit. the not rapidly changing image itself. expected and reasonable since noise is most "rapid" for a video encoder.

**Comment by Originalimoc** [comment]
@ReenigneArcher
One possibility just pops into my head when none of us thought before: framerate.

The TV noise example clip is just 30FPS and the screen capture API is producing the frame for the encoder at that rate too. If you play something like testufo and leave it at the screen corner so the capture is always at highest screen HZ(100/120/144/165 here I'm using), then this issue goes away(well, I set 25Mbps it gaves me a 30Mbps stream though). If the testufo is not put there only ~10Mbps will get used out of 25Mbps I set, leaving a much blurrier picture at 4K screen. There is no free compression lunch.

This issue is real and should be reopen and fix in the future.
